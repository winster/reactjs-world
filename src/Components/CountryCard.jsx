import React from 'react';
import {
    withRouter
  } from "react-router-dom";

class CountryCardComponent extends React.Component {
    handleOnClick = (countryCode) => {
        this.props.history.push("/"+countryCode);
    }
    render() {
        let country = this.props.data;
        country.population = country.population.toLocaleString('en');
        return (
        <div className="countrycard" onClick={this.handleOnClick.bind(this,country.alpha3Code)}>
            <div className="cardimage">
                <img src={country.flag} alt={country.name}/>
            </div>
            <div className="countrytext">
                <div className="countryname">
                    {country.name}
                </div>
                <div className="region countryelement">
                    <label>Population:</label> <span>{country.population}</span>
                </div>
                <div className="population countryelement">
                    <label>Region:</label> <span>{country.region}</span>
                </div>
                <div className="capital countryelement">
                    <label>Capital:</label> <span>{country.capital}</span>
                </div>
            </div>
        </div>
        );
    }
}

export default withRouter(CountryCardComponent);