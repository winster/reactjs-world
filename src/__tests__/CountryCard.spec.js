import React from "react";
import { create } from "react-test-renderer";
import CountryCard from '../Components/CountryCard';
import { mountWrap, shallowWrap } from '../testUtils/contextWrap';

describe("Country card component", () => {
    let country = {
        "alpha3Code":"AUS",
        "flag":"https://restcountries.eu/data/aut.svg",
        "name":"Austria",
        "population":"8725931",
        "region":"Europe",
        "capital":"Vienna",
        "numericCode": "040"
    };  
    let props = {data:country};
    let component;
    const wrappedShallow = () => shallowWrap(<CountryCard {...props} />);
  
    const wrappedMount = () => mountWrap(<CountryCard {...props} />);
  
    test("Matches the snapshot", () => {
        const wrapper = wrappedShallow();
        expect(wrapper).toMatchSnapshot();
    });
});