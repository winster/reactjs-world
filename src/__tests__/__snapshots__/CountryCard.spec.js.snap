// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country card component Matches the snapshot 1`] = `ShallowWrapper {}`;
